x-pack/legacy/plugins/integrations_manager/index.ts

```diff
@@ -64,17 +67,31 @@ const pluginOptions: LegacyPluginOptions = {
   },
   configPrefix: PLUGIN.CONFIG_PREFIX,
   publicDir: resolve(__dirname, 'public'),
-  config: undefined,
+  config(Joi: typeof JoiNamespace) {
+    return getConfigSchema(Joi);
+  },
   deprecations: undefined,
   preInit: undefined,
-  init(server: Server) {
+  // yes, any. See https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/infra/server/lib/adapters/configuration/kibana_configuration_adapter.ts#L49-L58
+  // for a way around it, but this is Legacy Platform and I'm not sure these hoops are worth jumping through.
+  init(server: any) {
     server.plugins.xpack_main.registerFeature(feature);
 
     // convert hapi instance to KbnServer
     // `kbnServer.server` is the same hapi instance
     // `kbnServer.newPlatform` has important values
     const kbnServer = (server as unknown) as KbnServer;
-    const initializerContext: PluginInitializerContext = {};
+
+    const getConfig$ = <T>() =>
+      defer<T>(async () => await server.config().get(PLUGIN.CONFIG_PREFIX));
+
+    const initializerContext = ({
+      config: {
+        create: getConfig$,
+        createIfExists: getConfig$,
+      },
+    } as unknown) as PluginInitializerContext;
```

Comment:
tying into @jfsiii 's comments about defining your own `PluginInitializerContext` just for the integrations_manager plugin. I would recommend you rather change your own type to match your shim than to cast to unknown here as you won't get any type errors from this.

One way to do it would be to only pick the properties you've already shimmed as you probably don't need to shim the entire PluginInitializerContext. Something like:
```ts
type IntegrationsPluginInitializerContext = Pick<PluginInitializerContext, 'config'>
```